std.fmt on comptime_int fails to handle some decimal values

`comptimePrint` doesn't handle correctly some decimal values (such as 300.0, but other are fine, such as 300.5).

How to replicate the issue:

```zig
const std = @import("std");


pub fn main() void {
    const value = 30.0;
    const value_to_str = std.fmt.comptimePrint("{d}", .{value});
    std.debug.print("value = {s}\n", .{value_to_str});
}
```

Expected output:

```
value = 30.0
```

Retrieved output:

```
An error occurred:
/usr/local/bin/lib/std/fmt.zig:1367:47: error: overflow of integer type 'u1' with value '2'
/usr/local/bin/lib/std/fmt.zig:802:27: note: called from here
/usr/local/bin/lib/std/fmt.zig:732:58: note: called from here
/usr/local/bin/lib/std/fmt.zig:487:31: note: called from here
/usr/local/bin/lib/std/fmt.zig:184:23: note: called from here
/usr/local/bin/lib/std/fmt.zig:2008:11: note: called from here
/usr/local/bin/lib/std/fmt.zig:2052:83: note: called from here
```

Code that works:
```zig
const std = @import("std");


pub fn main() void {
    const value = 30.1;
    const value_to_str = std.fmt.comptimePrint("{d}", .{value});
    std.debug.print("value = {s}\n", .{value_to_str});
}
```

Output:

```
value = 30.1
```

--- a/lib/std/fmt.zig
+++ b/lib/std/fmt.zig
@@ -1839,6 +1839,8 @@ test comptimePrint {
     try std.testing.expectEqualSlices(u8, "100", comptimePrint("{}", .{100}));
     try std.testing.expectEqualStrings("30", comptimePrint("{d}", .{30.0}));
     try std.testing.expectEqualStrings("30.0", comptimePrint("{d:3.1}", .{30.0}));
+    try std.testing.expectEqualStrings("0.05", comptimePrint("{d}", .{0.05}));
+    try std.testing.expectEqualStrings("5e-2", comptimePrint("{e}", .{0.05}));
 }
 
 test "parse u64 digit too big" {

--- a/lib/std/fmt/format_float.zig
+++ b/lib/std/fmt/format_float.zig
@@ -270,9 +270,9 @@ pub fn formatDecimal(buf: []u8, f_: FloatDecimal128, precision: ?usize) FormatEr
 
     // fixed bound: leading_digit(1) + point(1)
     const req_bytes = if (f.exponent >= 0)
-        2 + @abs(f.exponent) + olength + (precision orelse 0)
+        @as(usize, 2) + @abs(f.exponent) + olength + (precision orelse 0)
     else
-        2 + @max(@abs(f.exponent) + olength, precision orelse 0);
+        @as(usize, 2) + @max(@abs(f.exponent) + olength, precision orelse 0);
     if (buf.len < req_bytes) {
         return error.BufferTooSmall;
     }

